ast.ts

```diff
@@ -37,6 +39,8 @@ export type Expr<A> =
   | {  a?: A, tag: "lookup", obj: Expr<A>, field: string }
   | {  a?: A, tag: "method-call", obj: Expr<A>, method: string, arguments: Array<Expr<A>> }
   | {  a?: A, tag: "construct", name: string }
+  | {  a?: A, tag: "list_expr", contents: Array<Expr<A>> }
+  | {  a?: A, tag: "list_lookup", list: Expr<A>, index: Expr<A> }
```

Comment:
Note that this is going to have to parse exactly the same as this dict bracket lookup form from the dict group. These AST forms should be harmonized to something like `bracket-lookup` and `bracket-assign`, since we can't tell them apart at parse-time.

https://github.com/ucsd-cse231-w21/chocopy-wasm-compiler/pull/24/files#diff-08420cd45e72258159c0a8701934f95b0e70129064a4867914f3a3b5782a0e5fR41

@lite-david these groups should agree on these and we can merge what you decide on.

Reply:
@erauyeung @jpolitz  Let's settle on this?  `| {  a?: A, tag: "bracket-lookup", obj:Expr<A>, key:Expr<A> }`  @erauyeung

Reply:
@lite-david sure, we have updated our ast accordingly. For assignment, we have temporary change the ast to `| {  a?: A, tag: "bracket-assign", obj: Expr<A>, key: Expr<A>, value: Expr<A> }` before the new structure come out.